src/lib.rs

```diff
@@ -40,21 +40,24 @@ pub trait Body {
     type Error;
 
     /// Attempt to pull out the next data buffer of this stream.
-    fn poll_data(&mut self, cx: &mut Context<'_>) -> Poll<Option<Result<Self::Data, Self::Error>>>;
+    fn poll_data(
+        self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Option<Result<Self::Data, Self::Error>>>;
 
-    /// Returns the bounds on the remaining length of the stream.
-    ///
-    /// When the **exact** remaining length of the stream is known, the upper bound will be set and
-    /// will equal the lower bound.
-    fn size_hint(&self) -> SizeHint {
-        SizeHint::default()
-    }
+    // /// Returns the bounds on the remaining length of the stream.
+    // ///
+    // /// When the **exact** remaining length of the stream is known, the upper bound will be set and
+    // /// will equal the lower bound.
+    // fn size_hint(&self) -> SizeHint {
```

Comment:
Why is this commented out? `size_hint` is needed by hyper to determine the content length.